test/cql-pytest/test_describe.py

```diff
@@ -447,6 +447,49 @@ def test_desc_index(cql, test_keyspace):
         
         assert f"CREATE INDEX named_index ON {tbl}{maybe_space}(c)" in c_desc
 
+def test_desc_index_on_collections(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c frozen<set<int>>, d map<int, int>, e list<int>, f set<int>, g list<int>, h set<int>") as tbl:
+        indexes = [
+            ("idx_b",           f"CREATE INDEX idx_b ON {tbl}(b)"),
+            ("idx_c",           f"CREATE INDEX idx_c ON {tbl}(full(c))"),
+            ("idx_d_entries",   f"CREATE INDEX idx_d_entries ON {tbl}(entries(d))"),
+            ("idx_d_keys",      f"CREATE INDEX idx_d_keys ON {tbl}(keys(d))"),
+            ("idx_d_values",    f"CREATE INDEX idx_d_values ON {tbl}(values(d))"),
+            ("idx_e",           f"CREATE INDEX idx_e ON {tbl}(values(e))"),
+            ("idx_f",           f"CREATE INDEX idx_f ON {tbl}(values(f))"),
+        ]
+
+        for _, create_stmt in indexes:
+            cql.execute(create_stmt)
+        
+        for name, create_stmt in indexes:
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc
+        
+        # When an index is created on list column with `CREATE INDEX ON tbl(list_column)`,
+        # its target is `values(list_column)` and describe will print creat statement with that
+        # values() function added.
+        cql.execute(f"CREATE INDEX idx_g ON {tbl}(g)")
+        idx_g_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_g").one().create_statement
+        assert f"CREATE INDEX idx_g ON {tbl}(values(g))" in idx_g_desc
+
+        # The same situation but with set
+        cql.execute(f"CREATE INDEX idx_h ON {tbl}(h)")
+        idx_h_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_h").one().create_statement
+        assert f"CREATE INDEX idx_h ON {tbl}(values(h))" in idx_h_desc
+
+def test_desc_index_column_quoting(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, \"select\" int, \"some???name\" map<int, int>") as tbl:
+        indexes = [
+            ("idx_1", f"CREATE INDEX idx_1 ON {tbl}(\"select\")"),
+            ("idx_2", f"CREATE INDEX idx_2 ON {tbl}(keys(\"some???name\"))"),
+        ]
+
+        for name, create_stmt in indexes:
+            cql.execute(create_stmt)
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc
```

Comment:
I think (but correct me if I'm wrong) you don't have a test for local secondary index of collection, so please add it. Of course this will be `scylla_only` because local secondary index is a Scylla extension that doesn't exist in Cassandra.

Reply:
Added test for secondary indexes on regular column and on frozen collection column. Indexes on other column types like: set<>, list<> are not supported yet.

Also, I didn't created a test for quoting in secondary indexes because the code to generate description in Scylla is exactly the same.